lib/src/context.dart

```diff
@@ -0,0 +1,147 @@
+  Context(this.Provider, this.Consumer, this._jsThis);
+  final ReactContext _jsThis;
+
+  /// Every [Context] object comes with a Provider component that allows consuming components to subscribe
+  /// to context changes.
+  ///
+  /// Accepts a `value` prop to be passed to consuming components that are descendants of this [Provider].
+  final ReactJsContextComponentFactoryProxy Provider;
+
+  /// A React component that subscribes to context changes.
+  /// Requires a function as a child. The function receives the current context value and returns a React node.
+  final ReactJsContextComponentFactoryProxy Consumer;
+  ReactContext get jsThis => _jsThis;
+}
+
+/// Creates a [Context] object. When React renders a component that subscribes to this [Context]
+/// object it will read the current context value from the closest matching Provider above it in the tree.
+///
+/// The `defaultValue` argument is only used when a component does not have a matching [Context.Provider]
+/// above it in the tree. This can be helpful for testing components in isolation without wrapping them.
+///
+/// __Example__:
+///
+///     Context MyContext = createContext('test');
+///
+///     class MyContextTypeClass extends react.Component2 {
+///       @override
+///       final contextType = MyContext;
+///
+///       render() {
+///         return react.span({}, [
+///           '${this.context}', // Outputs: 'test'
+///         ]);
+///       }
+///     }
+///
+/// ___ OR ___
+///
+///     Context MyContext = createContext();
+///
+///     class MyClass extends react.Component2 {
+///       render() {
+///         return MyContext.Provider({'value': 'new context value'}, [
+///           MyContext.Consumer({}, (value) {
+///             return react.span({}, [
+///               '$value', // Outputs: 'new context value'
+///             ]),
+///           });
+///         ]);
+///       }
+///     }
+///
+/// Learn more: https://reactjs.org/docs/context.html#reactcreatecontext
+Context createContext([
+  dynamic defaultValue,
+  int Function(dynamic currentValue, dynamic nextValue) calculateChangedBits,
+]) {
+  int jsifyCalculateChangedBitsArgs(currentValue, nextValue) {
+    return calculateChangedBits(
+        ContextHelpers.unjsifyNewContext(currentValue), ContextHelpers.unjsifyNewContext(nextValue));
+  }
+
+  var JSContext = React.createContext(ContextHelpers.jsifyNewContext(defaultValue),
+      calculateChangedBits != null ? allowInterop(jsifyCalculateChangedBitsArgs) : null);
+  return Context(
+    new ReactJsContextComponentFactoryProxy(JSContext.Provider, isProvider: true),
+    new ReactJsContextComponentFactoryProxy(JSContext.Consumer, isConsumer: true),
+    JSContext,
+  );
+}
+
+// A JavaScript symbol that we use as the key in a JS Object to wrap the Dart.
+@JS()
+external get _reactDartContextSymbol;
+
+/// A context utility for to assist with common needs of ReactDartContext.
+///
+/// __For internal/advanced use only.__
+class ContextHelpers {
```

Comment:
#nit could be `abstract` since it only contains statics and should not be implemented
```suggestion
abstract class ContextHelpers {
```